sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/ShowCreateTableSuite.scala

```diff
@@ -0,0 +1,140 @@
+    val table = "ddl_test"
+    withTable(table) {
+      sql(
+        s"""CREATE TABLE $table
+           |USING json
+           |PARTITIONED BY (c)
+           |CLUSTERED BY (a) SORTED BY (b) INTO 2 BUCKETS
+           |COMMENT 'This is a comment'
+           |TBLPROPERTIES ('a' = '1')
+           |AS SELECT 1 AS a, "foo" AS b, 2.5 AS c
+         """.stripMargin
+      )
+      checkCreateTable(table)
+    }
+  }
+
+  protected def checkCreateTable(table: String, serde: Boolean = false): Unit = {
+    checkCreateTableOrView(TableIdentifier(table, Some("default")), "TABLE", serde)
+  }
+
+  protected def checkCreateView(table: String, serde: Boolean = false): Unit = {
+    checkCreateTableOrView(TableIdentifier(table, Some("default")), "VIEW", serde)
+  }
+
+  protected def checkCreateTableOrView(
+      table: TableIdentifier,
+      checkType: String,
+      serde: Boolean): Unit = {
+    val db = table.database.getOrElse("default")
+    val expected = spark.sharedState.externalCatalog.getTable(db, table.table)
+    val shownDDL = if (serde) {
+      sql(s"SHOW CREATE TABLE ${table.quotedString} AS SERDE").head().getString(0)
+    } else {
+      sql(s"SHOW CREATE TABLE ${table.quotedString}").head().getString(0)
+    }
+    sql(s"DROP $checkType ${table.quotedString}")
+    try {
+      sql(shownDDL)
+      val actual = spark.sharedState.externalCatalog.getTable(db, table.table)
+      checkCatalogTables(expected, actual)
+    } finally {
+      sql(s"DROP $checkType IF EXISTS ${table.table}")
+    }
+  }
+
+  protected def checkCatalogTables(expected: CatalogTable, actual: CatalogTable): Unit = {
+    assert(CatalogTable.normalize(actual) == CatalogTable.normalize(expected))
+  }
+}
+
+/**
+ * The class contains tests for the `SHOW CREATE TABLE` command to check V1 In-Memory
+ * table catalog.
+ */
+class ShowCreateTableSuite extends ShowCreateTableSuiteBase with CommandSuiteBase {
+  override def commandVersion: String = super[ShowCreateTableSuiteBase].commandVersion
+
+  test("SHOW CREATE TABLE") {
+    val t = "tbl"
+    withTable(t) {
+      sql(
+        s"""
+           |CREATE TABLE $t (
+           |  a bigint NOT NULL,
+           |  b bigint,
+           |  c bigint,
+           |  `extraCol` ARRAY<INT>,
+           |  `<another>` STRUCT<x: INT, y: ARRAY<BOOLEAN>>
+           |)
+           |$defaultUsing
+           |OPTIONS (
+           |  from = 0,
+           |  to = 1,
+           |  via = 2)
+           |COMMENT 'This is a comment'
+           |TBLPROPERTIES ('prop1' = '1', 'prop2' = '2', 'prop3' = 3, 'prop4' = 4)
+           |PARTITIONED BY (a)
+           |LOCATION '/tmp'
+        """.stripMargin)
+      val showDDL = getShowCreateDDL(s"SHOW CREATE TABLE $t")
+      assert(showDDL === Array(
+        s"CREATE TABLE `default`.`$t` (",
+        "`b` BIGINT,",
+        "`c` BIGINT,",
+        "`extraCol` ARRAY<INT>,",
+        "`<another>` STRUCT<`x`: INT, `y`: ARRAY<BOOLEAN>>,",
+        "`a` BIGINT NOT NULL)",
```

Comment:
This looks like a bug in v1, as Spark silently changes the table schema from what the user specified.

Reply:
@Peng-Lei can you confirm with the real table schema? maybe it's just a bug in SHOW CREATE TABLE.

Reply:
ok.

Reply:
@cloud-fan I add code after create table
```scala
      val actual = spark.sharedState.externalCatalog.getTable("default", t)
      println(actual)
```

The result is as follows:
```scala
CatalogTable(
Database: default
Table: tbl
Created Time: Tue Nov 30 03:34:05 PST 2021
Last Access: UNKNOWN
Created By: Spark 3.3.0-SNAPSHOT
Type: EXTERNAL
Provider: parquet
Comment: This is a comment
Table Properties: [prop1=1, prop2=2, prop3=3, prop4=4]
Location: file:/tmp
Storage Properties: [from=0, to=1, via=2]
Partition Provider: Catalog
Partition Columns: [`a`]
Schema: root
 |-- b: long (nullable = true)
 |-- c: long (nullable = true)
 |-- extraCol: array (nullable = true)
 |    |-- element: integer (containsNull = true)
 |-- <another>: struct (nullable = true)
 |    |-- x: integer (nullable = true)
 |    |-- y: array (nullable = true)
 |    |    |-- element: boolean (containsNull = true)
 |-- a: long (nullable = false)
)
```

Reply:
Do you think this is a bug in v1 with wrong order of columns in schema? If so, I want to try to resolve it.

Reply:
yea I think so

Reply:
@cloud-fan This problem may be a bit slower to process for me, I want to process this PR first and I also add todo in the code.